### ISSUE TITLE
Why are the results slower?

### ISSUE THREAD

**Issue by Nick088Official** [haoheliu/versatile_audio_super_resolution#52]
I tested a bit with audio sr and noticed that the audio is longer, slowed, and pitched down.

Heres some tests i did:
[audio sr low quality tests.zip](https://github.com/haoheliu/versatile_audio_super_resolution/files/15299251/audio.sr.low.quality.tests.zip)

It's kinda weird, could somebody have an answer/explaination for this? And if i could avoid it? Idk if its a problem on my end

**Comment by lustfeind** [comment]
Still no fix for this?

**Comment by Nick088Official** [comment]
Guess not, tried contacting their discord server but seems kinda dead

**Comment by FlexTestHD** [comment]
I have the same issue and am reassured that the problem is not unique to me. The online demo on Replicate works without any problems.

**Comment by Nick088Official** [comment]
That's weird, i tried running on colab audio sr and had this issue, you may think the code is a bit different in the Replicate demo?

**Comment by FlexTestHD** [comment]
I would be very surprised if the code there is different. From another GitHub post about a fork of that (Audio-Upscaler), I read from a user (dillfrescott) that this problem with the pitch occurs when the sampling rates are mismatched.

**Comment by FlexTestHD** [comment]
Short update: The problem with the pitch only seems to occur when using the gradio demo (at least for me). The problem does not occur when using the command line interface.

**Comment by Nick088Official** [comment]
> I would be very surprised if the code there is different. From another GitHub post about a fork of that (Audio-Upscaler), I read from a user (dillfrescott) that this problem with the pitch occurs when the sampling rates are mismatched.


> Short update: The problem with the pitch only seems to occur when using the gradio demo (at least for me). The problem does not occur when using the command line interface.

So is the problem about mismatched sampling rates or the gradio UI at the end? see if that user can reply to you

**Comment by FlexTestHD** [comment]
Honestly, I think it's the Gradio UI. With the exact same audio files, upscaling via CLI works fine for me, whereas when using the WebUI there is the pitch difference. I therefore do not think that it is a sampling mismatch, then the problem should have to occur in the same way when using the CLI. Have you tried it without the WebUI or the Replicate demo with your audio files yet?

**Comment by Nick088Official** [comment]
> Have you tried it without the WebUI or the Replicate demo with your audio files yet?

Yes, i just tried with an audio sr no ui colab i made (that will probably will be released publicly), and works all fine, i will see if maybe i can fix the gradio uu someway or atleast check the code

**Comment by Nick088Official** [comment]
> Honestly, I think it's the Gradio UI. With the exact same audio files, upscaling via CLI works fine for me, whereas when using the WebUI there is the pitch difference. I therefore do not think that it is a sampling mismatch, then the problem should have to occur in the same way when using the CLI. 

I just looked at the code and for some reason in the gradio ui one it takes the waveform at 44100 hz, but AudioSR should be doing an Output of 48k Hz, so i just changed that value and it's not slowed down anymore (https://github.com/Nick088Official/versatile_audio_super_resolution/commit/ca8086259d724a31ac52ac900c1cdadfa1a0b3c7) , i don't know much if it loses still some quality compared to the CLI one, only weird thing is it adds like 2 seconds of silence at the end for some reason, might be a thing of gradio.

Also here are some tests of original low quality vs cli one vs gradio 44100 vs gradio 48khz: [audio sr results test 2.zip](https://github.com/user-attachments/files/16059499/audio.sr.results.test.2.zip)
